### PR TITLE
Improve ordinal and clock notation hot paths

### PR DESCRIPTION
**Pull request by clairernovotny**
## Summary
- avoid eager allocation/formatting in ordinal, clock-notation, transformer, metric, phrase-join, and single-part TimeSpan hot paths
- preserve existing public API behavior with focused tests around optimized branches, culture-sensitive formatting, and boundary cases
- address Copilot review feedback for culture-specific ordinal formatting, adjacent-hyphen normalization, and context-sensitive title casing
- cache default ordinal number formatting by culture name to remove the large `int.Ordinalize(culture)` positive-number regression found in the benchmark artifacts
- repair benchmark comparison reporting so ResultsComparer restores with an isolated NuGet config and no longer uploads a hidden build-failure fragment as a successful report
- expand BenchmarkDotNet coverage for optimized paths and control paths:
  - metric default int small/kilo/mega plus boundary, giga, formatted fallback, and double milli cases
  - formatter first-touch/date/data-unit cases plus single-part, multi-part, zero, and words TimeSpan humanization

## Benchmark Evidence
Local net10 short BenchmarkDotNet runs on Apple M2 Max after `b959f088`:
- `MetricNumeralBenchmarks`: small `59.05 ns / 56 B`, kilo `117.68 ns / 112 B`, mega `113.29 ns / 112 B`, boundary `149.74 ns / 128 B`, giga `158.14 ns / 152 B`, formatted fallback `205.21 ns / 128 B`, milli `125.57 ns / 136 B`
- `FormatterBenchmarks`: Russian single-part TimeSpan `95.84 ns / 64 B`, multi-part control `423.60 ns / 600 B`, zero `161.50 ns / 80 B`, words `131.71 ns / 224 B`, Romanian Date `140.24 ns / 136 B`, Arabic DataUnit `39.56 ns / 80 B`

Short local review-fix spot checks after `58084558`:
- `OrdinalBenchmarks`: English ordinalize `29.93 ns / 64 B`; custom-culture paths unchanged in shape for Dutch/Turkish/Greek/Finnish control cases
- `TransformersBenchmarks`: title case 10 chars `201.24 ns / 696 B`, 100 chars `2.424 us / 7264 B`, 1000 chars `24.483 us / 73200 B`

Short local ordinal regression check after `3221a0ad`:
- `OrdinalBenchmarks`: English `27.63 ns / 64 B`, Dutch `44.51 ns / 176 B`, Turkish `45.18 ns / 176 B`, Greek `46.37 ns / 176 B`, Finnish `44.53 ns / 176 B`
- The prior GitHub artifact for `58084558` showed Dutch/Turkish/Greek around `110-111 us`; the cache change removes that regression locally.

GitHub benchmark workflow:
- `b959f088` completed successfully: https://github.com/Humanizr/Humanizer/actions/runs/24616985676
- `58084558` completed successfully, but its comparison artifact exposed the ResultsComparer restore/reporting bug: https://github.com/Humanizr/Humanizer/actions/runs/24617740031
- latest head `3221a0ad` is running from the push and PR events:
  - https://github.com/Humanizr/Humanizer/actions/runs/24618892125
  - https://github.com/Humanizr/Humanizer/actions/runs/24618892631

## Tests
- `dotnet build src/Benchmarks/Benchmarks.csproj -c Release -f net10.0`
- `dotnet run --project src/Benchmarks/Benchmarks.csproj -c Release -f net10.0 -- --filter '*MetricNumeralBenchmarks*' --job short --warmupCount 1 --iterationCount 3`
- `dotnet run --project src/Benchmarks/Benchmarks.csproj -c Release -f net10.0 -- --filter '*FormatterBenchmarks*' --job short --warmupCount 1 --iterationCount 3`
- `dotnet run --project src/Benchmarks/Benchmarks.csproj -c Release -f net10.0 -- --filter '*OrdinalBenchmarks*' --job short --warmupCount 1 --iterationCount 3`
- `dotnet run --project src/Benchmarks/Benchmarks.csproj -c Release -f net10.0 -- --filter '*TransformersBenchmarks*' --job short --warmupCount 1 --iterationCount 3`
- `dotnet test --project tests/Humanizer.Tests/Humanizer.Tests.csproj --framework net10.0 --no-restore -- --filter-class '*OrdinalizeTests' --filter-class '*TransformersTests' --filter-class '*CoverageGapTests'`
- `dotnet test --project tests/Humanizer.Tests/Humanizer.Tests.csproj --framework net10.0 --no-restore -- --filter-class '*OrdinalizeTests' --filter-class '*DutchGenderedOrdinalTests'`
- `dotnet test --project tests/Humanizer.Tests/Humanizer.Tests.csproj --framework net11.0 --no-restore -- --filter-class '*OrdinalizeTests' --filter-class '*DutchGenderedOrdinalTests'`
- `dotnet run --project src/Benchmarks/Benchmarks.csproj -c Release -f net10.0 -- --filter '*OrdinalBenchmarks*' --job short --warmupCount 1 --iterationCount 1`
- `dotnet format Humanizer.slnx --verify-no-changes --verbosity diagnostic --no-restore`
- `ruby -e 'require "yaml"; YAML.load_file(".github/workflows/benchmarks-baseline-vs-current.yml"); puts "yaml ok"'`
- `dotnet pack src/Humanizer/Humanizer.csproj -c Release -o artifacts/packages/ordinal-benchmark-fix`
- `dotnet test --project tests/Humanizer.Tests/Humanizer.Tests.csproj --framework net10.0 --no-restore`
- `dotnet test --project tests/Humanizer.Tests/Humanizer.Tests.csproj --framework net11.0 --no-restore`
- `git diff --check`

Note: `dotnet test --project tests/Humanizer.Tests/Humanizer.Tests.csproj --framework net8.0 --no-restore` could not run locally because this machine does not have the .NET 8 runtime installed; CI should cover it.

### CODE REVIEW

File: src/Humanizer/OrdinalizeExtensions.cs [hunk 1]
```diff
@@ -246,9 +248,90 @@ public static string Ordinalize(this int number, GrammaticalGender gender, Cultu
...
+    {
+        if (string.IsNullOrEmpty(culture.Name))
+        {
+            return CreateOrdinalNumberFormatting(culture);
+        }
+
+        return OrdinalNumberFormattingCache.GetOrAdd(culture.Name, static cultureName =>
+            CreateOrdinalNumberFormatting(CultureInfo.GetCultureInfo(cultureName)));
+    }
+
```
[the remark this answers is not in the document]
Comment (clairernovotny) [reply src/Humanizer/OrdinalizeExtensions.cs:285]: I checked this path rather than adding the suggested comparison because reading `culture.NumberFormat` on every positive ordinal call is the cold-path cost this PR is avoiding (locally ~90 us/op for `new CultureInfo("nl").NumberFormat`). For positive `int.ToString(...)`, the cited `NativeDigits` / `DigitSubstitution` customizations do not change the formatted digits on the supported modern .NET path; I also verified a cloned culture with Arabic-Indic native digits still formats `123` as `123`, matching the cached default behavior. Negative-sign customizations remain instance-aware in `GetOrdinalNumberFormatting`, so the observable custom-format behavior this path affects is still covered without reintroducing the regression.
